server/sonar-server/src/main/java/org/sonar/server/authentication/UserIdentityAuthenticator.java

```diff
@@ -172,4 +209,8 @@ private void removeGroups(DbSession dbSession, UserDto userDto, Collection<Strin
     return organizationFlags.isEnabled(dbSession) ? Optional.empty() : Optional.of(defaultGroupFinder.findDefaultGroup(dbSession, defaultOrganizationProvider.get().getUuid()));
   }
 
+  private static UserDto[] toArray(Optional<UserDto> userDto) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Specify a "UserDto" parameter instead. ([squid:S3553](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2FMMF-1161%2Fallow_auth_of_same_email&issues=AWFVrGmdsS4rYBZqFOJN&open=AWFVrGmdsS4rYBZqFOJN)

[ISSUE_KEY]: AWFVrGmdsS4rYBZqFOJN